modules/services/way-displays.nix

```diff
@@ -0,0 +1,112 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    attrsets
+    lists
+    literalExpression
+    maintainers
+    mkEnableOption
+    mkIf
+    mkOption
+    types
+    ;
+  mergeSets = sets: lists.fold attrsets.recursiveUpdate { } sets;
+  cfg = config.services.way-displays;
+  yaml = pkgs.formats.yaml { };
+in
+{
+  meta.maintainers = [ maintainers.jolars ];
+
+  options.services.way-displays = {
+    enable = mkEnableOption "way-displays";
+
+    package = lib.mkPackageOption pkgs "way-displays" { };
+
+    settings = mkOption {
+      type = yaml.type;
+      default = { };
+      example = literalExpression ''
+        {
+          ORDER = [
+            "DP-2"
+            "Monitor Maker ABC123"
+            "!^my_regex_here[0-9]+"
+            "'!.*$'"
+          ];
+          SCALING = false;
+          MODE = [
+            {
+              NAME_DESC = "HDMI-A-1";
+              WIDTH = 1920;
+              HEIGHT = 1080;
+              HZ = 60;
+            }
+          ];
+          TRANSFORM = [
+            {
+              NAME_DESC = "eDP-1"
+              TRANSFORM = "FLIPPED-90";
+            }
+          ];
+        };
+      '';
+      description = ''
+        The way-displays configuration written to
+        {file}`$XDG_CONFIG_HOME/way-displays/cfg.yml`. See
+        <https://github.com/alex-courtis/way-displays/wiki/Configuration> for a
+        description of available options.
+      '';
+    };
+
+    systemdTarget = mkOption {
+      type = types.str;
+      default = config.wayland.systemd.target;
+      defaultText = literalExpression "config.wayland.systemd.target";
+      description = ''
+        Systemd target to bind to.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.way-displays" pkgs lib.platforms.linux)
+    ];
+
+    xdg.configFile."way-displays/cfg.yaml".source =
+      yaml.generate "way-displays-config.yaml"
+        (mergeSets [
+          {
+            CALLBACK_CMD = "${pkgs.libnotify}/bin/notify-send \"way-displays \${CALLBACK_LEVEL}\" \"\${CALLBACK_MSG}\"";
+          }
```

Comment:
Looks like you're always including this regardless of the user configuration, this feels like it should be optional since it includes another dependency, as well.

Reply:
> The reason I included this is that this is the default setting, but it is not guaranteed to work out-of-the-box if notify-send isn't in the path. I guess we could patch the package in nixpkgs too, but this seemed like an easier solution (albeit with the alternative maintenance load of having to keep the default up to date).

I see what you mean with this being a default upstream https://github.com/alex-courtis/way-displays/blob/master/examples/cfg.yaml#L78 If this will just be used to `fix` the derivation not wrapping `libnotify` upstream, then it makes more sense.

Reply:
Might make sense then, to make this a `mkDefault` though so user's wanting to set their own CALLBACK_CMD don't have to mkForce it.